### PR TITLE
[ci-skip] workflow for alpine-utils:latest image

### PR DESCRIPTION
**Pull request by RaviKiranJanjanam**
Signed-off-by: RaviKiranJanjanam <ravi.kiran.janjanam@accenture.com>

**Changelog**

Added the workflow for alpine-image:latest
fixed the repository name to be in lowercase for below files under workflow

.github/workflows/corda_images_build.yml
.github/workflows/docker_build.yml
.github/workflows/indy_image_build.yaml
.github/workflows/quorum_images_build.yml
modified platforms/hyperledger-besu/configuration/roles/create/helm_component/templates/crypto_tessera.tpl to include alpne-utils with latest tag instead of 1.1 

 

**Reviewed by**
@sownak @suvajit-sarkar @jagpreetsinghsasan 

 

**Linked issue**
#2147

### CODE REVIEW

File: .github/workflows/alpine_image.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        string: ${{ env.IMAGE_PATH }}
+
+    - name: Build and push release tag latest
+      id: alpine_build_latest
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: ./platforms/shared/images/alpine-utils.Dockerfile
+        push: true
+        tags: ${{ steps.image_path_format.outputs.lowercase }}/alpine-utils:1.1
```
Comment (suvajit-sarkar) [inline .github/workflows/alpine_image.yaml:63]: the tag should be based on the release tag for releases, latest for develop branch push

File: .github/workflows/alpine_image.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+##############################################################################################
+#  Copyright Accenture. All Rights Reserved.
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+# Build bevel-build docker image triggered by :
```
Comment (suvajit-sarkar) [inline .github/workflows/alpine_image.yaml:7]: Change comments

**Review by suvajit-sarkar — CHANGES_REQUESTED** [review]

**Comment by suvajit-sarkar** [comment]
Kindly check the acceptance criterion https://github.com/hyperledger/bevel/issues/2147

File: platforms/hyperledger-besu/configuration/roles/create/helm_component/templates/crypto_tessera.tpl [hunk 3]
```diff
@@ -28,7 +28,7 @@ spec:
       # alpineutils image has the binaries like jq,curl, wget and openssl in it. 
       # having this image with binaries helps in removing the dependencies of utility binaries in the besu and tessera containers
       # dockerfile can be found at shared/images/apline-utils.Dockerfile
-      alpineutils: {{ network.docker.url }}/alpine-utils:1.1
+      alpineutils: "{{ network.docker.url }}/alpine-utils:{{ network.bevel_version | default('latest') }}"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-besu/configuration/roles/create/helm_component/templates/crypto_tessera.tpl:31]: This should not be changed as `network.bevel_version` gives a wrong sense of information that by changing to lets say, `1.0` version, will indeed make use of Bevel 1.0 code.
Bevel versions are heavily affected by code changes and the right way to use a specific version of Bevel is via changing the `tag` of the local Bevel repo (from where the deployments are made).

File: .github/workflows/alpine_image.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+# Build apline-utils docker image triggered by :
+#  - push on develop branch for latest tags 
+#  - create tag vx.y.z for release tags
+#  - Can be triggered manually
+
+name: Build alpine1.1 Docker image and push
```
Comment (jagpreetsinghsasan) [inline .github/workflows/alpine_image.yaml:12]: This should be renamed to `Build alpine docker image and push to ghcr`

File: .github/workflows/alpine_image.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        string: ${{ env.IMAGE_PATH }}
+
+    - name: Build and push release tag latest
+      id: alpine_build_latest
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: ./platforms/shared/images/alpine-utils.Dockerfile
+        push: true
+        tags: ${{ steps.image_path_format.outputs.lowercase }}/alpine-utils:latest
```
Comment (jagpreetsinghsasan) [inline .github/workflows/alpine_image.yaml:66]: We should not use `latest` tag here and specifically mention the `1.1` version

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
1. Is this tested for all the platforms? (As this will affect them all)
2. This PR should change all instances of alpine image usages in all platforms to strictly use `1.1` instead of `1.0` or `latest`
3. What is env.IMAGE_PATH in this (exact value)? (To understand why are we making it lowercase everywhere)

**Comment by suvajit-sarkar** [comment]
I'm resolving the conversation and closing this PR

**Pull request merged**